packages/main/src/SegmentedButton.js

```diff
@@ -129,15 +129,15 @@ class SegmentedButton extends UI5Element {
 	}
 
 	onBeforeRendering() {
-		let length = 0,
-			buttons = this.getSlottedNodes("buttons");
+		let length = 0;
+		const buttons = this.getSlottedNodes("buttons");
 
 		buttons.forEach((item, index) => {
 			item.posInSet = index + 1;
 			length += 1;
 		});
 
-		buttons.forEach((item) => {
+		buttons.forEach((item, index) => {
```

Comment:
don't need a second loop to assign the set size. You just need to declare a variable for the buttons length and use it the first loop. Then, you can remove the second loop as you won't need it anymore.
```js
// with a variable and single loop
const buttonsCount/buttonsLength/buttonsNumber = buttons.length;
```
```js
buttons.forEach((item, index) => {
			item.posInSet = index + 1;
			item.sizeOfSet = buttonsLength;
});
```

or

```js
// even without a variable and single loop
buttons.forEach((item, index, arr) => {
			item.posInSet = index + 1;
			item.sizeOfSet = arr.length;
});
```